File: frontend/src/application/use-cases/users/update-user-status.js
import { IUserRepository } from '../../../domain/repositories/user-repository.js';

/**
 * Caso de uso para actualizar el estado (activo/inactivo) de un usuario en el frontend.
 */
export class UpdateUserStatusUseCase {
  /**
   * @param {IUserRepository} userRepository - Una implementación del repositorio de usuarios.
   */
  constructor(userRepository) {
    if (!(userRepository instanceof IUserRepository)) {
      throw new Error('userRepository must be an instance of IUserRepository');
    }
    this.userRepository = userRepository;
  }

  /**
   * Ejecuta el caso de uso para actualizar el estado de un usuario.
   * @param {number} userId - El ID del usuario.
   * @param {boolean} isActive - El nuevo estado de activación.
   * @returns {Promise<object>} Una promesa que resuelve con el objeto del usuario actualizado o un mensaje de éxito.
   */
  async execute(userId, isActive) {
    return this.userRepository.updateUserStatus(userId, isActive);
  }
}

File: backend/users/application/use_cases/update_user_status.py
from typing import Optional, Dict, Any
from ...domain.repositories.user_repository import IUserRepository
from ...models import User

class UpdateUserStatusUseCase:
    """
    Caso de uso para actualizar el estado (activo/inactivo) de un usuario.
    Esta clase reside en la capa de Aplicación.
    """
    def __init__(self, user_repository: IUserRepository):
        self.user_repository = user_repository

    async def execute(self, user_id: int, is_active: bool) -> Optional[User]:
        # El caso de uso UpdateUserProfileUseCase podría usarse para esto si se pasa {'is_active': is_active}
        # Pero crear un caso de uso específico para cambiar el estado puede ser más claro.
        user_data_to_update: Dict[str, Any] = {'is_active': is_active}
        return await self.user_repository.update(user_id, user_data_to_update)
